nhost/service/manager.go

```diff
@@ -86,13 +95,162 @@ func (m *dockerComposeManager) SetGitBranch(gitBranch string) {
 	m.composeConfig = compose.NewConfig(m.nhostConfig, m.ports, m.env, gitBranch, m.projectName)
 }
 
+func (m *dockerComposeManager) checkPorts(ctx context.Context) error {
+	select {
+	case <-ctx.Done():
```

Comment:
for a boolean check it is much easier to do `if ctx.Error() == context.Canceled { return nil }`

Reply:
I haven't tested the method myself but I feel that it might be a bit slow and the context cancellation seems to be ignore for most of it, breaking this down into smaller chunks that check the context might help you verify the context more often